sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -163,7 +164,13 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       //      other choice.
       case j @ ExtractEquiJoinKeys(joinType, leftKeys, rightKeys, nonEquiCond, left, right, hint) =>
         def createBroadcastHashJoin(onlyLookingAtHint: Boolean) = {
-          getBroadcastBuildSide(left, right, joinType, hint, onlyLookingAtHint, conf).map {
+          val buildSide = getBroadcastBuildSide(
```

Comment:
can we make it more general? I think we can also give a warning if SMJ/SHJ hint is specified but the join has no join keys.